src/client/MealsComponents/MealDetail.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import { MakeReservation } from "../ReservationComponents/Reservation";
+
+function MealsListDetails({ fetchMeal }) {
+  if (fetchMeal) {
+    const detailsOfMeals = fetchMeal.map((meal) => (
+      <li key={meal.id} className="mealdetailsSection">
+        <Link to={`/meals/${meal.id}`}>
+          <h3>{meal.title}</h3>
+        </Link>
+
+        <p>
+          Description : {meal.description}
+          <br></br>
+          Price : {meal.price} Kr
+          <br></br>
+          Location: {meal.location}
+          <br></br>
+          Max Reservation : {meal.max_reservations}
+          <br></br>
+          Meal_id : {meal.id}
+        </p>
+      </li>
+    ));
+    console.log(fetchMeal);
+
+    return <ul className="mealList">{detailsOfMeals}</ul>;
+  }
+}
+
+export function MealsListInformation({ match }) {
+  const [fetchMeal, setFetchMeal] = useState([]);
+  useEffect(() => {
+    fetchMealResult();
+    console.log(match);
+  }, []);
+  const fetchMealResult = async () => {
+    const response = await fetch(
+      `http://localhost:3000/api/meals/${match.params.id}`
+    );
+    const Data = await response.json();
+    setFetchMeal(Data);
+  };
```

Comment:
```suggestion
  useEffect(() => {
    const fetchMealResult = async () => {
      const response = await fetch(
        `http://localhost:3000/api/meals/${match.params.id}`
      );
      const Data = await response.json();
      setFetchMeal(Data);
    };
    
    fetchMealResult();
  }, []);
```

Prefer this pattern, of putting the effect function inside of the `useEffect` block. It makes it more clear that it's only used as part of the effect and it puts the logic of the effect up front to be read and understood.

Reply:
I have changed this section of code and it works fines. it renders the data very nicely